src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,245 @@
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String query = "INSERT INTO cars (model, manufacturer_id) VALUES(?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createCarStatement =
+                        connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            createCarStatement.setString(1, car.getModel());
+            createCarStatement.setLong(2, car.getManufacturer().getId());
+            createCarStatement.executeUpdate();
+            ResultSet resultSet = createCarStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                Long id = resultSet.getObject(1, Long.class);
+                car.setId(id);
+            }
+
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't create car. " + car, e);
+        }
+
+        insertDrivers(car);
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String selectRequest = "SELECT c.id, c.model, m.id as manufacturer_id, m.name, m.country"
+                + " FROM taxi.cars c "
+                + "JOIN taxi.manufacturers m "
+                + "ON c.manufacturer_id = m.id where c.id = ? and c.is_deleted = false;";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                    PreparedStatement getStatement =
+                            connection.prepareStatement(selectRequest)) {
+            getStatement.setLong(1, id);
+            ResultSet resultSet = getStatement.executeQuery();
+            if (resultSet.next()) {
+                car = parseCarWithDriverResultSet(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get a car with id " + id, e);
+        }
+        if (car != null) {
+            car.setDrivers(getDriversForCar(id));
+        }
+        return Optional.of(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String selectRequest = "SELECT c.id, c.model, m.id as manufacturer_id, m.name, m.country"
+                + " FROM taxi.cars c "
+                + "JOIN taxi.manufacturers m "
+                + "ON c.manufacturer_id = m.id where c.is_deleted = false;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getStatement =
+                        connection.prepareStatement(selectRequest)) {
+            ResultSet resultSet = getStatement.executeQuery();
+            while (resultSet.next()) {
+                Car car = parseCarWithDriverResultSet(resultSet);
+                if (car != null) {
+                    car.setDrivers(getDriversForCar(car.getId()));
+                }
```

Comment:
here you will open several connections inside this already opened connection
please move this below try-with-resources